### PR TITLE
Add crypto, update module name and version.

### PR DESCRIPTION
**Pull request by JonathanHenson**
Sha256, Md5, SHA256 HMAC

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awscrt/crypto.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @staticmethod
+    def md5_new():
+        return Hash(native_handle=_aws_crt_python.aws_py_md5_new())
+
+    def update(self, to_hash):
+        _aws_crt_python.aws_py_hash_update(self._hash, to_hash)
+
+    def digest(self, truncate_to = 0):
+        tmp = _aws_crt_python.aws_py_hash_digest(self._hash, truncate_to)
+        return tmp
```
Comment (ColdenCullen) [inline awscrt/crypto.py]: why tmp?

File: awscrt/crypto.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class Hash(object):
+    def __init__(self, native_handle):
+        self._hash = native_handle
+
+    @staticmethod
+    def sha256_new():
+        return Hash(native_handle=_aws_crt_python.aws_py_sha256_new())
+
+    @staticmethod
+    def md5_new():
```
Comment (ColdenCullen) [inline awscrt/crypto.py]: I feel like these are confusing... We should probably remove the `_new`.

File: awscrt/crypto.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import _aws_crt_python
+
+class Hash(object):
```
Comment (ColdenCullen) [inline awscrt/crypto.py]: The way this class is used is kind of confusing, can you document it please?

File: awscrt/crypto.py (hunk 1 above)
Comment (JonathanHenson) [reply awscrt/crypto.py]: doh! that was testing.

**Pull request merged**